bucket/bucket_plain.go

```diff
@@ -16,34 +15,34 @@ type Plain struct {
 }
 
 // NewPlain builds and returns new Plain instance
-func NewPlain(section string, operation MetricOperation, success bool) *Plain {
+func NewPlain(section string, operation MetricOperation, success, uniDecode bool) *Plain {
 	operationSanitized := make([]string, cap(operation))
 	for k, v := range operation {
-		operationSanitized[k] = SanitizeMetricName(v)
+		operationSanitized[k] = SanitizeMetricName(v, uniDecode)
 	}
-	return &Plain{SanitizeMetricName(section), strings.Join(operationSanitized, "."), success}
+	return &Plain{SanitizeMetricName(section, uniDecode), strings.Join(operationSanitized, "."), success}
 }
 
 // Metric builds simple metric name in the form:
 //  <section>.<operation-0>.<operation-1>.<operation-2>
 func (b *Plain) Metric() string {
-	return fmt.Sprintf("%s.%s", b.section, b.operation)
+	return b.section + "." + b.operation
 }
 
 // MetricWithSuffix builds metric name with success suffix in the form:
 //  <section>-ok|fail.<operation-0>.<operation-1>.<operation-2>
 func (b *Plain) MetricWithSuffix() string {
-	return fmt.Sprintf("%s-%s.%s", b.section, getOperationStatus(b.success), b.operation)
+	return b.section + "-" + operationsStatus[b.success] + "." + b.operation
 }
 
 // MetricTotal builds simple total metric name in the form:
 //  total.<section>
 func (b *Plain) MetricTotal() string {
-	return fmt.Sprintf("%s.%s", totalBucket, b.section)
+	return totalBucket + "." + b.section
 }
 
 // MetricTotalWithSuffix builds total metric name with success suffix in the form
 //  total-ok|fail.<section>
 func (b *Plain) MetricTotalWithSuffix() string {
-	return fmt.Sprintf("%s.%s-%s", totalBucket, b.section, getOperationStatus(b.success))
+	return totalBucket + "." + b.section + "-" + operationsStatus[b.success]
```

Comment:
If you use `bytes.Buffer` here you can even save more allocations

Reply:
0 allocations here =)